frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/moderation/ModerationPortlet.java

```diff
@@ -803,4 +839,104 @@ private String getDefaultLicenseInfoHeaderText() {
+        List<ModerationRequest> sortedModerationRequests = sortModList(moderationList, paginationParameters);
+        int count = PortletUtils.getProjectDataCount(paginationParameters, moderationList.size());
+        User user = UserCacheHolder.getUserFromRequest(request);
+        boolean isClearingAdmin = PermissionUtils.isUserAtLeast(UserGroup.CLEARING_ADMIN, user);
+        JSONArray moderationRequestData = createJSONArray();
+        for (int i = paginationParameters.getDisplayStart(); i < count; i++) {
+            JSONObject jsonObject = JSONFactoryUtil.createJSONObject();
+            ModerationRequest modreq = sortedModerationRequests.get(i);
+            jsonObject.put("id", modreq.getId());
+            jsonObject.put("renderTimestamp", modreq.getTimestamp() + "");
+            jsonObject.put("componentType", printEnumValueWithTooltip(request, modreq.getComponentType()));
+            jsonObject.put("documentName", modreq.getDocumentName());
+            jsonObject.put("requestingUser", UserUtils.displayUser(modreq.getRequestingUser(), null));
+            jsonObject.put("requestingUserDepartment", modreq.getRequestingUserDepartment());
+            jsonObject.put("moderators", displayUserCollection(modreq.getModerators()));
+            jsonObject.put("moderationState", printEnumValueWithTooltip(request, modreq.getModerationState()));
+            if (!open) {
+                jsonObject.put("isClearingAdmin", isClearingAdmin);
+            }
+            moderationRequestData.put(jsonObject);
+        }
+
+        return moderationRequestData;
+    }
+
+    private List<ModerationRequest> sortModList(List<ModerationRequest> modList, PaginationParameters modParameters) {
+        boolean isAsc = modParameters.isAscending().orElse(true);
+
+        switch (modParameters.getSortingColumn().orElse(MODERATION_DT_ROW_TIMESTAMP)) {
+        case MODERATION_DT_ROW_TIMESTAMP:
+            Collections.sort(modList, PortletUtils.compareByTimeStamp(isAsc));
+            break;
+        case MODERATION_DT_ROW_COMPONENT_TYPE:
+            Collections.sort(modList, PortletUtils.compareByComponentType(isAsc));
+            break;
+        case MODERATION_DT_ROW_DOCUMENT_NAME:
+            Collections.sort(modList, PortletUtils.compareByDocumentName(isAsc));
+            break;
+        case MODERATION_DT_ROW_REQUESTING_USER:
+            Collections.sort(modList, PortletUtils.compareByRequestingUser(isAsc));
+            break;
+        case MODERATION_DT_ROW_REQUESTING_USER_DEPARTMENT:
+            Collections.sort(modList, PortletUtils.compareByRequestingUserDepartment(isAsc));
+            break;
+        case MODERATION_DT_ROW_MODERATION_STATE:
+            Collections.sort(modList, PortletUtils.compareByModerationState(isAsc));
+            break;
+        default:
+            break;
+        }
+
+        return modList;
+    }
+
+    private String displayUserCollection(Set<String> values) {
```

Comment:
Is it possible to handle this in UI like using tag `<sw360:DisplayUserEmailCollection>` and `<sw360:DisplayEnum>` ?

Reply:
Not sure whether it is possible to handle this in UI using tags <sw360:DisplayUserEmailCollection> and <sw360:DisplayEnum>.
Since tags are part of jsp and converted once on page loading. Moderation Datatable is ajax and js based.
However I tried it, but without positive outcome.
Please do suggest if you have a solution